### ISSUE TITLE
 core/image.cpp:1596 - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).

### ISSUE THREAD

**Issue by git2013vb** [godotengine/godot#36456]
**Godot version:**
3.2 stable mono
**OS/device including version:**
Debian 10
**Issue description:**
I have this error in Output dock:
```
Godot Engine v3.2.stable.mono.official (c) 2007-2020 Juan Linietsky, Ariel Manzur & Godot Contributors.
--- GDScript language server started ---
Switch Scene Tab
 core/image.cpp:1596 - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
 Width of image must be greater than 0.
 core/image.cpp:2376 - Condition "data.size() == 0" is true.
 drivers/png/png_driver_common.cpp:56 - png_image_write_to_memory: invalid argument
 drivers/png/png_driver_common.cpp:183 - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
 Can't convert image to PNG.
 editor/editor_node.cpp:5268 - An EditorPlugin build callback failed.
 Another resource is loaded from path 'res://Main.cs' (possible cyclic resource inclusion).
 Script does not inherit a Node: res://auto_load/Log.cs.
 editor/editor_autoload_settings.cpp:460 - Condition "!info->node" is true. Continuing.
```
EDIT.: I think it is related that Godot don't update the .csproj file - wrong log? -
 was missing Log.cs reference in:

```
<ItemGroup>
    <Compile Include="auto_load\LoadPCK.cs" />
    <Compile Include="auto_load\Log.cs" />
    <Compile Include="Main.cs" />
    <Compile Include="Properties\AssemblyInfo.cs" />
  </ItemGroup>
```

**Steps to reproduce:**
I added some script in auto load as singletons then these errors when I start the project - Log.cs.

**Minimal reproduction project:**
[desert_edge_server_sharp2.zip](https://github.com/godotengine/godot/files/4239469/desert_edge_server_sharp2.zip)

**Comment by vicguedez** [comment]
I'm not sure if related to this issue, but I got today this:

```
ERROR: create: Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
   At: core/image.cpp:1596
ERROR: Width of image must be greater than 0.
   At: core/image.cpp:1103
ERROR: lock: Condition "data.size() == 0" is true.
   At: core/image.cpp:2376
WARNING: check_error: png_image_write_to_memory: invalid argument
     At: drivers/png/png_driver_common.cpp:56
ERROR: image_to_png: Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
   At: drivers/png/png_driver_common.cpp:183
ERROR: Can't convert image to PNG.
   At: drivers/png/resource_saver_png.cpp:56
```

Today I reimported all the pngs as "Video RAM" instead of "Lossless", then the game started freezing intermitently while loading a scene. I saw this error, opened the scene in editor and just save it without modification, I think the error is gone now.

It can be useful to print out the file in with the issue in those errors.

**Comment by avencherus** [comment]
With a 3.2.2 custom build, today I got these errors.  Will be setting a break point and will see if I can get a stack trace.

Also just happened to see them in someone's video just the other day.  They were new to Godot and thought you had to restart the editor.

https://youtu.be/vxRzLf4PdgY?t=988

```
**ERROR**: Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
   At: core\image.cpp:1603:Image::create() - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
**ERROR**: Width of image must be greater than 0.
   At: core\image.cpp:1108:Image::crop_from_point() - Condition "p_width <= 0" is true.
**ERROR**: Condition "data.size() == 0" is true.
   At: core\image.cpp:2389:Image::lock() - Condition "data.size() == 0" is true.
**WARNING**: png_image_write_to_memory: invalid argument
   At: drivers\png\png_driver_common.cpp:56:PNGDriverCommon::check_error() - png_image_write_to_memory: invalid argument
**ERROR**: Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
   At: drivers\png\png_driver_common.cpp:183:PNGDriverCommon::image_to_png() - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
**ERROR**: Can't convert image to PNG.
   At: drivers\png\resource_saver_png.cpp:56:ResourceSaverPNG::save_image() - Condition "err" is true. Returned: err
**ERROR**: Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
   At: core\image.cpp:1603:Image::create() - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
**ERROR**: Width of image must be greater than 0.
   At: core\image.cpp:1108:Image::crop_from_point() - Condition "p_width <= 0" is true.
**ERROR**: Condition "data.size() == 0" is true.
   At: core\image.cpp:2389:Image::lock() - Condition "data.size() == 0" is true.
**WARNING**: png_image_write_to_memory: invalid argument
   At: drivers\png\png_driver_common.cpp:56:PNGDriverCommon::check_error() - png_image_write_to_memory: invalid argument
**ERROR**: Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
   At: drivers\png\png_driver_common.cpp:183:PNGDriverCommon::image_to_png() - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
**ERROR**: Can't convert image to PNG.
   At: drivers\png\resource_saver_png.cpp:56:ResourceSaverPNG::save_image() - Condition "err" is true. Returned: err
**ERROR**: Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
   At: core\image.cpp:1603:Image::create() - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
**ERROR**: Width of image must be greater than 0.
   At: core\image.cpp:1108:Image::crop_from_point() - Condition "p_width <= 0" is true.
**ERROR**: Condition "data.size() == 0" is true.
   At: core\image.cpp:2389:Image::lock() - Condition "data.size() == 0" is true.
**WARNING**: png_image_write_to_memory: invalid argument
   At: drivers\png\png_driver_common.cpp:56:PNGDriverCommon::check_error() - png_image_write_to_memory: invalid argument
**ERROR**: Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
   At: drivers\png\png_driver_common.cpp:183:PNGDriverCommon::image_to_png() - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
**ERROR**: Can't convert image to PNG.
   At: drivers\png\resource_saver_png.cpp:56:ResourceSaverPNG::save_image() - Condition "err" is true. Returned: err
**ERROR**: Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
   At: core\image.cpp:1603:Image::create() - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
**ERROR**: Width of image must be greater than 0.
   At: core\image.cpp:1108:Image::crop_from_point() - Condition "p_width <= 0" is true.
**ERROR**: Condition "data.size() == 0" is true.
   At: core\image.cpp:2389:Image::lock() - Condition "data.size() == 0" is true.
**WARNING**: png_image_write_to_memory: invalid argument
   At: drivers\png\png_driver_common.cpp:56:PNGDriverCommon::check_error() - png_image_write_to_memory: invalid argument
**ERROR**: Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
   At: drivers\png\png_driver_common.cpp:183:PNGDriverCommon::image_to_png() - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
**ERROR**: Can't convert image to PNG.
   At: drivers\png\resource_saver_png.cpp:56:ResourceSaverPNG::save_image() - Condition "err" is true. Returned: err
**ERROR**: Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
   At: core\image.cpp:1603:Image::create() - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
**ERROR**: Width of image must be greater than 0.
   At: core\image.cpp:1108:Image::crop_from_point() - Condition "p_width <= 0" is true.
**ERROR**: Condition "data.size() == 0" is true.
   At: core\image.cpp:2389:Image::lock() - Condition "data.size() == 0" is true.
**WARNING**: png_image_write_to_memory: invalid argument
   At: drivers\png\png_driver_common.cpp:56:PNGDriverCommon::check_error() - png_image_write_to_memory: invalid argument
**ERROR**: Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
   At: drivers\png\png_driver_common.cpp:183:PNGDriverCommon::image_to_png() - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
**ERROR**: Can't convert image to PNG.
   At: drivers\png\resource_saver_png.cpp:56:ResourceSaverPNG::save_image() - Condition "err" is true. Returned: err
```

**Comment by adam-bates** [comment]
+1 with errors in output

**Godot version:**
3.2.2.stable.official
**OS/device including version:**
Macbook Pro (2020, 13") running MacOS Catalina (version 10.15.5 (19F101))

---

Also, issue can be seen repro'd in [HeartBeast's tutorial on YouTube](https://youtu.be/srQz4Ix8rZM?t=89). I think he's running a Windows desktop.

**Comment by RichmondGame** [comment]
Got the following error when I created a New Inherited Scene. The scene only had a Node as the root, with a gd script attached.
 
core/image.cpp:1604 - Index p_width - 1 = -1 is out of bounds (MAX_WIDTH = 16384).
 Width of image must be greater than 0.
 core/image.cpp:2390 - Condition "data.size() == 0" is true.
 drivers/png/png_driver_common.cpp:56 - png_image_write_to_memory: invalid argument
 drivers/png/png_driver_common.cpp:183 - Condition "compressed_size <= png_size_estimate" is true. Returned: FAILED
 Can't convert image to PNG.

**Comment by volzhs** [comment]
AFAIK, it happens when saving a newly created scene with script editor open.

**Comment by qarmin** [comment]
Godot try to get scene texture, but when this scene was never opened in 2D viewport, then Godot never generates texture and return image with 0 height and 0 width

https://github.com/godotengine/godot/blob/af35d0d3c83161848fad952f24d42592b1a9b932/editor/editor_node.cpp#L1124

Errors comes later when trying to format and save this texture

Steps to reproduce
- Open 2 scenes with Node2D as root
- In editor check settings to restore all opened scenes after closing
- Open editor - there should open first scene
- Go to script editor
- Choose second scene from scene tabs
- Try to save it



Possible solution:
If returned texture from `scene_root->get_texture()->get_data();` have 0 pixels, then we could just jump over image processing code.

**Comment by akien-mga** [comment]
Fixed by #40163.
